### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName: regression / false positive

### ISSUE THREAD

**Issue by lars-sh** [pmd/pmd#2098]
https://github.com/pmd/pmd/commit/493e0014b042bf1076244ad7b7c7e91a31f52d42#diff-bc84f21dff8725b7dc1441ccbc43bef1R243-R253

**Affects PMD Version:** *6.18.0 and 6.19.0*

**Rule:** [UnnecessaryFullyQualifiedName](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)

**Description:**

*I do have a package-info.java file containing an annotation that lays in a subpackage. Therefore specifying a fully qualified name is required. Though, the rule `UnnecessaryFullyQualifiedName` outputs a violation.*

*This problem does **not** occur with PMD 6.17.0. I took a look at the rules code and at a first view https://github.com/pmd/pmd/commit/493e0014b042bf1076244ad7b7c7e91a31f52d42#diff-bc84f21dff8725b7dc1441ccbc43bef1R243-R253 seems to cause the problem.*

**Code Sample demonstrating the issue:**

```
@com.example.subpackage.ExampleAnnotation
package com.example;
```

*`pmd.xml` example:*

```
<file name="...\com\example\package-info.java">
<violation beginline="1" endline="1" begincolumn="2" endcolumn="42" rule="UnnecessaryFullyQualifiedName" ruleset="Code Style" externalInfoUrl="https://pmd.github.io/pmd-6.18.0/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname" priority="4">
Unnecessary use of fully qualified name 'com.example.subpackage.ExampleAnnotation' due to existing same package import 'com.example.*'
</violation>
</file>
```

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
@lars-sh just out of curiosity which Maven target are you using? `pmd:pmd` / `pmd:check`? or `verify`?

The Maven plugin doesn't really use type resolution unless the `verify` target is used as classes are not compiled (@adangel is this something we can fix in the plugin?)

**Comment by lars-sh** [comment]
Many thanks @adangel 👍 .

I hope it's OK, if I'm not verifying that fix, but awaiting the next release.

**Comment by Han-Cui** [comment]
If the "subpackage" in the code sample "@com.example.subpackage.ExampleAnnotation" here **just a name of class**  instead of a real subpackage, in com.example, will the fix make a False Negative? If so, pmd should raise a warning, I think. How to distinguish whether it is a subpackage?
